istioctl/cmd/auth_test.go

```diff
@@ -20,10 +20,24 @@ import (
 	"strings"
 	"testing"
 
+	"istio.io/istio/istioctl/pkg/auth"
+
 	"istio.io/istio/pilot/test/util"
 )
 
-func runCommandAndCheckGolden(name, command, golden string, t *testing.T) {
+func runCommandAndCheckGoldenFile(name, command, golden string, t *testing.T) {
+	out := runCommand(name, command, t)
+	util.CompareContent(out.Bytes(), golden, t)
+}
+
+func runCommandAndCheckExpectedString(name, command, expected string, t *testing.T) {
+	out := runCommand(name, command, t)
+	if out.String() != expected {
+		t.Errorf("test %q failed. \nExpected\n%s\nGot%s\n", name, expected, out.String())
+	}
```

Comment:
Should we use util.Compare() instead of directly comparing the two strings?

Reply:
I think we're using == to compare strings across Istio codebase. Are you worried about performance? fwiw, `strings.Compare()` uses == internally

Reply:
Oh I think you are right. util.Compare is to compare raw bytes. Ignore my comment here.